sencha-workspace/SlateTasksTeacher/app/controller/Dashboard.js

```diff
@@ -1,43 +1,457 @@
     },
 
 
     // controller configuration
     views: [
-        'Dashboard'
+        'Dashboard',
+        'TaskEditor',
+        'TaskRater',
+        'TaskAssigner'
+    ],
+    stores: [
+        'CourseSections',
+        'Students',
+        'StudentTasks',
+        'Tasks'
     ],
-
     refs: {
         dashboardCt: {
             selector: 'slate-tasks-teacher-dashboard',
             autoCreate: true,
 
             xtype: 'slate-tasks-teacher-dashboard'
         },
-        taskGrid: 'slate-tasks-teacher-dashboard slate-tasks-teacher-studentstaskgrid'
-    },
+        tasksGrid: {
+            selector: 'slate-studentsgrid',
+            autoCreate: true,
 
+            xtype: 'slate-studentsgrid'
+        },
+        taskAssigner: {
+            selector: 'slate-tasks-teacher-taskassigner',
+            autoCreate: true,
+
+            xtype: 'slate-tasks-teacher-taskassigner'
+        },
+        taskRater: {
+            selector: 'slate-tasks-teacher-taskrater',
+            autoCreate: true,
+
+            xtype: 'slate-tasks-teacher-taskrater'
+        },
+        taskEditor: {
+            selector: 'slate-tasks-teacher-taskeditor',
+            autoCreate: true,
+
+            xtype: 'slate-tasks-teacher-taskeditor'
+        },
+        taskEditorForm: 'slate-tasks-teacher-taskeditor slate-modalform',
+        skillsField: 'slate-tasks-teacher-taskeditor slate-skillsfield',
+        attachmentsField: 'slate-tasks-teacher-taskeditor slate-tasks-attachmentsfield',
+        assignmentsField: 'slate-tasks-teacher-taskeditor slate-tasks-assignmentsfield',
+        assignmentsComboField: 'slate-tasks-teacher-taskeditor slate-tasks-assignmentsfield combo',
+
+        courseSelector: 'slate-tasks-teacher-appheader combo'
+    },
 
     // controller templates method overrides
     onLaunch: function () {
         this.getDashboardCt().render('slateapp-viewport');
     },
 
-
     // event handlers
-    onCompetencyRowClick: function(me, competency, ev, targetEl) {
-        me.toggleCompetency(competency);
+    onCourseSectionSelect: function(combo, record) {
+        var me = this;
+
+        me.getDashboardCt().setCourseSection(record);
+    },
+
+    onDashboardSectionChange: function(record) {
+        this.redirectTo('section/'+record.get('Code'));
+    },
+
+    onAcceptTaskClick: function() {
+        var me = this,
+            taskRater = me.getTaskRater(),
+            studentTask = taskRater.getStudentTask();
+
+        studentTask.set('TaskStatus', 'completed');
+
+        me.saveStudentTask(studentTask, function() {
+            taskRater.close();
+        });
+    },
+
+    onAssignRevisionClick: function(btn) {
+        var me = this,
+            coords = btn.getXY(),
+            datepicker;
+
+        if (btn.dateSelected) {
+            me.assignStudentTaskRevision(btn.dateSelected);
+        } else {
+            datepicker = Ext.widget({
+                xtype: 'datepicker',
+                floating: true,
```

Comment:
@nbey this controller needs to explicitly depend on the datepicker component via its `requires` array.

Also, you can use the ref system to cleanly define a "factory" for these pickers:

```javascript
refs: {
    datePicker: {
        forceCreate: true,

        xtype: 'datepicker',
        floating: true
    }
}
```
(I'm not sure if a `selector` is still required, it would be useless with `forceCreate` but give the framework what it wants if it complains about it being missing)

and then you can do:
```
me.getDatePicker({
    handler: function(picker, date) {
        // ...
    }
});
```